app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -87,8 +87,18 @@ def sync_queue(auth_user = nil)
   end
 
   def playbooks_in_git_repository
-    git_repository.update_repo
-    git_repository.entries(scm_branch, "").grep(/\.ya?ml$/)
+    [].tap do |playbooks|
+      git_repository.update_repo
+      git_repository.with_worktree do |worktree|
+        worktree.ref = scm_branch
+        worktree.blob_list do |root, entry|
+          playbook_filepath = Pathname.new("#{root}#{entry[:name]}")
+          next if not_playbook_dir?(playbook_filepath)
```

Comment:
not sure I like the negative method name.  Might be better as `next unless playbook_dir?`

Reply:
For what it is worth, I was trying to match the `skip_directory` logic of `awx` as close as I could, hence why this originally ended up being what it was.

With the specs in place though, inverting the logic isn't that hard to do.